runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java

```diff
@@ -395,4 +495,55 @@ public static Color colorLerp(Color a, Color b, double t)
 			(int) Math.round(b1 + (t * (b2 - b1)))
 		);
 	}
+
+	/**
+	 * Fills all non-transparent pixels of the given image with the given color.
+	 *
+	 * @param image The image which should have its non-transparent pixels filled.
+	 * @param color The color with which to fill pixels.
+	 * @return      The given image with all non-transparent pixels set to the given color.
+	 */
+	public static BufferedImage getFilledImage(final BufferedImage image, final Color color)
+	{
+		return getFilledImage(image, color, SwingUtil::isNotFullyTransparent);
+	}
+
+	/**
+	 * 	Fills pixels of the given image with the given color based on a given fill condition
+	 * 	predicate.
+	 *
+	 * @param image         The image which should have its non-transparent pixels filled.
+	 * @param color         The color with which to fill pixels.
+	 * @param fillCondition The condition on which to fill pixels with the given color.
+	 * @return              The given image with all pixels fulfilling the fill condition predicate
+	 *                      set to the given color.
+	 */
+	public static BufferedImage getFilledImage(final BufferedImage image, final Color color, final Predicate<Color> fillCondition)
+	{
+		final BufferedImage filledImage = new BufferedImage(image.getWidth(), image.getHeight(), BufferedImage.TYPE_INT_ARGB);
+		for (int x = 0; x < filledImage.getWidth(); x++)
+		{
+			for (int y = 0; y < filledImage.getHeight(); y++)
+			{
+				final Color pixelColor = new Color(image.getRGB(x, y));
+				if (!fillCondition.test(pixelColor))
+				{
+					continue;
+				}
+
+				filledImage.setRGB(x, y, color.getRGB());
+			}
+		}
+		return filledImage;
+	}
+
+	private static boolean isFullyTransparent(final Color color)
```

Comment:
These 2 below methods seem unnecessary

Reply:
These were written so I wouldn't have to use unnamed default values in the `outlineImage` methods. I can see removing `isFullyTransparent` and moving its logic into `isNotFullyTransparent`, but I'm not sure of a more graceful way to set those default values.